client/state/sharing/publicize/publicize-actions/actions.js

```diff
@@ -104,3 +107,37 @@ export function editPostShareAction( siteId, postId, actionId, message, share_da
 		} );
 	};
 }
+
+export function schedulePostShareAction( siteId, postId, message, share_date, connections ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: PUBLICIZE_SHARE_ACTION_SCHEDULE,
+			siteId,
+			postId,
+			connections,
+		} );
+
+		return Promise.all(
+			connections.map( connection_id => wpcom.req.post( {
+				path: `/sites/${ siteId }/posts/${ postId }/publicize/scheduled-actions/`,
+				body: { message, share_date, connection_id },
+				apiNamespace: 'wpcom/v2',
+			} ) )
+		)
+			.catch( error => dispatch( {
+				type: PUBLICIZE_SHARE_ACTION_SCHEDULE_FAILURE,
+				siteId,
+				postId,
+				error,
+				connections
+			} ) )
+			.then( data => dispatch( {
```

Comment:
Minor, but `data` could be changed to `items` here so we could use the shorthand object property below.